### ISSUE TITLE
missing doctring false positive

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#281]
Originally reported by: **Pedro Algarvio (BitBucket: [s0undt3ch](http://bitbucket.org/s0undt3ch), GitHub: @s0undt3ch?)**

---

Consider the following hypothetical module

``` python
# -*- coding: utf-8 -*-
'''
Module docstring
'''

DEFAULT_LOCATION = 'here'


def get_location_1(opts, provider=None):
    '''
    Return the region to use, in this order:
        opts
        location
        default location ({0})
    '''.format(DEFAULT_LOCATION)
    return opts.get(
        'location', provider.get(
            'location', DEFAULT_LOCATION
        )
    )


def get_location_2(opts, provider=None):
    '''
    Return the region to use, in this order:
        opts
        location
        default location (FOO)
    '''
    return opts.get(
        'location', provider.get(
            'location', DEFAULT_LOCATION
        )
    )
```

And now consider pylint's output

```
$ pylint --reports=no /tmp/f.py 
************* Module f
C:  9, 0: Missing function docstring (missing-docstring)
```

The only different is that the first function has a `.format()` call. It **has** a docstring.

```
$ pylint --version
pylint 1.2.1, 
astroid 1.1.1, common 0.61.0
Python 2.7.8 (default, Jul  1 2014, 17:30:21) 
[GCC 4.9.0 20140604 (prerelease)]
```

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/281
